app/controllers/pages_controller.rb

```diff
@@ -35,7 +35,44 @@ def create
   end
 
   def show
-    render_liquid(@page.liquid_layout, :show)
+    if one_click?
+      process_one_click
+      flash[:notice] = t('fundraiser.thank_you_with_amount', amount: "#{view_context.number_to_currency(params[:amount], unit: params[:currency])}")
+      redirect_to PageFollower.new_from_page(@page, member_id: recognized_member.id).follow_up_path
+    else
+      render_liquid(@page.liquid_layout, :show)
+    end
+  end
+
+  def one_click?
+    params.fetch(:one_click, '') == 'true' &&
+      payment_method_id &&
+      params.fetch(:amount, 0).to_f > 0 &&
+      params[:currency].present?
+  end
+
+  def process_one_click
+    opts = ActionController::Parameters.new(
+      payment: {
+        payment_method_id: payment_method_id,
+        currency: params[:currency],
+        amount: params[:amount]
+      },
+      user: {
+        email: recognized_member.email
+      },
+      page_id: @page.id
+    )
+
+    PaymentProcessor::Braintree::OneClick.new(opts).run
+  end
+
+  def payment_method_id
+    if current_member
+      current_member.customer.payment_methods.first.id
+    else
+      Payment::Braintree::PaymentMethod.find_by(token: (cookies.signed[:payment_methods] || '').split(',').first)
+    end
```

Comment:
@osahyoun I believe this should all go to a service that handles the one_click processing.